File: CVcodes/erode.cpp
#include "opencv2/highgui/highgui.hpp"
#include "opencv2/imgproc/imgproc.hpp"
#include <stdio.h>

using namespace cv;
using namespace std;

int erode_image();
int dilate_image();

int main()
{	
	erode_image();
	dilate_image();	
	return 0;
}
	
int erode_image()
{
	Mat img, kernel, fin_img;
	img = imread("Lenna.png");
	int size=4;
	kernel = getStructuringElement(MORPH_CROSS, Size(2*size+1, 2*size+1), Point(size, size));
	erode(img, fin_img, kernel);
	imshow("erosion", fin_img);
	waitKey(0);
	return 1;
}

int dilate_image()
{
	Mat img, kernel, fin_img;
	img = imread("Lenna.png");
	int size=4;
	kernel = getStructuringElement(MORPH_CROSS, Size(2*size+1, 2*size+1), Point(size, size));
	dilate(img, fin_img, kernel);
	imshow("dilate", fin_img);
	waitKey(0);
	return 1;
}

File: CVcodes/erode.py
import cv2

def dilate_image():
	img = cv2.imread("Lenna.png")
	size = 4
	kernel = cv2.getStructuringElement(cv2.MORPH_CROSS, (2*size+1, 2*size+1))
	fin_img = cv2.dilate(img, kernel)
	cv2.imshow("dialte",fin_img)
	cv2.waitKey(0)
	
def erode_image():
	img = cv2.imread("Lenna.png")
	size = 4
	kernel = cv2.getStructuringElement(cv2.MORPH_CROSS, (2*size+1, 2*size+1))
	fin_img = cv2.erode(img, kernel)
	cv2.imshow("dialte",fin_img)
	cv2.waitKey(0)
	
dilate_image()
erode_image()
